Author: Luca Boccassi <bluca@debian.org>
Description: reorder libraries and build eal before cmdline
 Most libraries and PMDs depend on eal, and eal depends only on kvargs,
 so reorder the list in Meson to reflect this and take advantage of this
 dependency chain.
Origin: https://patches.dpdk.org/patch/49997/
--- a/lib/meson.build
+++ b/lib/meson.build
@@ -9,9 +9,10 @@
 # given as a dep, no need to mention ring. This is especially true for the
 # core libs which are widely reused, so their deps are kept to a minimum.
 libraries = [ 'compat', # just a header, used for versioning
-	'cmdline', # ethdev depends on cmdline for parsing functions
 	'kvargs', # eal depends on kvargs
-	'eal', 'ring', 'mempool', 'mbuf', 'net', 'ethdev', 'pci', # core
+	'eal', # everything depends on eal
+	'cmdline', # ethdev depends on cmdline for parsing functions
+	'ring', 'mempool', 'mbuf', 'net', 'ethdev', 'pci', # core
 	'metrics', # bitrate/latency stats depends on this
 	'hash',    # efd depends on this
 	'timer',   # eventdev depends on this
